core-bundle/src/Resources/contao/library/Contao/System.php

```diff
@@ -329,6 +329,25 @@ public static function getContainer()
 	 */
 	public static function setContainer(ContainerInterface $container)
 	{
+		// Reset state if a container was previously set
+		if (null !== static::$objContainer)
+		{
+			foreach (self::$arrSingletons as $singleton)
+			{
+				if (\in_array('reset', get_class_methods($singleton), true))
+				{
+					$singleton::reset();
+				}
+			}
+
+			self::$removedServiceIds = null;
+			self::$arrStaticObjects = array();
+			self::$arrSingletons = array();
+			self::$arrLanguages = array();
+			self::$arrLanguageFiles = array();
+			self::$arrImageSizes = array();
```

Comment:
Can we be sure that this is not a BC break? `System::setContainer()` is used in the production code as well.

Reply:
It's a behavior change but also a bug to me. If you're re-setting the container to a new instance we should clear the related state. But maybe it would be better to test `$container !== static::$objContainer` instead, so we're gracefully ignoring if the same instance is set multiple times in a row.

Btw.: I think we're ever setting `System::setContainer` only once in production code anyway, do we?